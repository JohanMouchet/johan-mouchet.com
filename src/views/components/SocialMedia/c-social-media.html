{##
 # Social Media
 #
 # @param {array}  medias 		       - List of medias
 # @param {string} medias[0]name     - media name
 # @param {string} medias[0]URL      - media URL
 # @param {string} medias[0]iconPath - media iconPath
 #}

{% macro default(options) %}
	<div class="grid">
		{% for media in options.medias %}
			<div class="c-social-media cell cell--6-@xs cell--4-@sm cell--12-@md">
				<a class="c-social-media__link" href="{{ media.URL }}" title="Follow me on {{ media.name }}" target="_blank">
					{{ file_get_contents(media.iconPath)|raw }}{{ media.name }}
				</a>
			</div>
		{% endfor %}
	</div>
{% endmacro %}
