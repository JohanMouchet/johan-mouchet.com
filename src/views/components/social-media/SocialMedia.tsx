import {
  Icon500px,
  IconCodepen,
  IconDribbble,
  IconGithub,
  IconInstagram,
  IconLinkedIn,
  IconSpotify,
  IconTwitter,
  IconUnsplash,
  IconYoutube,
} from "@/views/objects/icons";
import clsx, { ClassValue } from "clsx";
import styles from "./SocialMedia.module.scss";

export const SocialMedia = ({
  media,
  className,
  ...props
}: {
  media: {
    name: string;
    link: {
      url: string;
    };
    icon:
      | "500px"
      | "codepen"
      | "dribbble"
      | "github"
      | "instagram"
      | "linkedin"
      | "spotify"
      | "twitter"
      | "unsplash"
      | "youtube";
  }[];
  className?: ClassValue;
} & Omit<React.HTMLProps<HTMLDivElement>, "media">) => {
  if (!media?.length) {
    return null;
  }

  const getIcon = (icon: string) => {
    if (icon === "500px") {
      return <Icon500px />;
    } else if (icon === "codepen") {
      return <IconCodepen />;
    } else if (icon === "dribbble") {
      return <IconDribbble />;
    } else if (icon === "github") {
      return <IconGithub />;
    } else if (icon === "instagram") {
      return <IconInstagram />;
    } else if (icon === "linkedin") {
      return <IconLinkedIn />;
    } else if (icon === "spotify") {
      return <IconSpotify />;
    } else if (icon === "twitter") {
      return <IconTwitter />;
    } else if (icon === "unsplash") {
      return <IconUnsplash />;
    } else if (icon === "youtube") {
      return <IconYoutube />;
    }
  };

  return (
    <div className={clsx("grid", className)} {...props}>
      {media.map((media) => (
        <div
          className={clsx(
            styles["c-social-media"],
            "cell",
            "cell-6",
            "sm:cell-4",
            "md:cell-12"
          )}
          key={media.link.url}
        >
          <a className={styles["c-social-media__link"]} href={media.link.url}>
            <span aria-hidden>{getIcon(media.icon)}</span>
            {media.name}
          </a>
        </div>
      ))}
    </div>
  );
};
