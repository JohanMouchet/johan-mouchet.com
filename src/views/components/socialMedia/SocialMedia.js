// @flow

import * as React from "react";
import "./social-media.scss";

type Props = {
  medias: [
    {
      name: string,
      URL: string,
      iconPath: string
    }
  ]
};

const SocialMedia = (props: Props) => {
  const { prop1, prop2 } = props;

  return (
    <div className="grid">
      {props.medias.map(media => (
        <div className="c-social-media cell cell--6-@xs cell--4-@sm cell--12-@md">
          <a
            className="c-social-media__link"
            href={media.URL}
            title="Follow me on { media.name }"
            target="_blank"
          >
            {media.iconPath}
            {media.name}
          </a>
        </div>
      ))}
    </div>
  );
};

SocialMedia.defaultProps = {
  prop1: 2
};

export default SocialMedia;
