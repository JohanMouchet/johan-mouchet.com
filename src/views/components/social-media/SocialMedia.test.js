import React from "react";
import { shallow } from "enzyme";
import SocialMedia from ".";

it("Renders <SocialMedia /> without crashing", () => {
  shallow(<SocialMedia />);
});
