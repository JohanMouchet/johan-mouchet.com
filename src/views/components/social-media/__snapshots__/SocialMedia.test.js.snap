// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <SocialMedia /> without crashing 1`] = `""`;
